src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Repositories/GuestFeedbackRepository.cs

```diff
@@ -0,0 +1,89 @@
+using System.Linq.Expressions;
+using AirBnB.Domain.Constants;
+using AirBnB.Domain.Entities;
+using AirBnB.Domain.Settings;
+using AirBnB.Persistence.Caching.Brokers;
+using AirBnB.Persistence.Caching.Models;
+using AirBnB.Persistence.DataContexts;
+using AirBnB.Persistence.Repositories.Interfaces;
+using Microsoft.Extensions.Options;
+
+namespace AirBnB.Persistence.Repositories;
+
+public class GuestFeedbackRepository(AppDbContext dbContext, ICacheBroker cacheBroker, 
+    IOptions<GuestFeedbacksCacheSettings> feedbackCachingSettings) 
+    : EntityRepositoryBase<GuestFeedback, AppDbContext>(dbContext, cacheBroker, new CacheEntryOptions()),
+        IGuestFeedbackRepository
+{
+    private readonly GuestFeedbacksCacheSettings _feedbackCachingSettings = feedbackCachingSettings.Value;
```

Comment:
Why do we need it from external configuration ?

Reply:
Guest feedback is stored in the cache using two methods. The first one is the default, which independently stores it for 5 minutes. The other method is for storing feedback specifically for listing rating processing service in a collection. Therefore, this second method should retain the feedback for a longer duration, unless the background service initiates and retrieves it from the cache. I am considering that if we modify the background service's execution time interval based on our user count, we should also adjust the cache expiration settings for guest feedback to prevent unexpected issues.